libraries/rush-lib/src/utilities/test/objectUtilities.test.ts

```diff
@@ -0,0 +1,166 @@
+
+      expect(objectsAreDeepEqual({}, {})).toEqual(true);
+      expect(objectsAreDeepEqual({}, undefined)).toEqual(false);
+      expect(objectsAreDeepEqual({}, null)).toEqual(false);
+      expect(objectsAreDeepEqual(undefined, {})).toEqual(false);
+      expect(objectsAreDeepEqual(null, {})).toEqual(false);
+
+      expect(objectsAreDeepEqual({ a: 1 }, { a: 1 })).toEqual(true);
+      expect(objectsAreDeepEqual({ a: 1 }, { a: 2 })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: 1 }, {})).toEqual(false);
+      expect(objectsAreDeepEqual({}, { a: 1 })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: 1 }, { b: 1 })).toEqual(false);
+
+      expect(objectsAreDeepEqual({ a: 1, b: 2 }, { a: 1, b: 2 })).toEqual(true);
+      expect(objectsAreDeepEqual({ a: 1, b: 2 }, { a: 1, b: 3 })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: 1, b: 2 }, { a: 1, c: 2 })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: 1, b: 2 }, { b: 2, a: 1 })).toEqual(true);
+    });
+
+    it('can compare nested objects', async () => {
+      const { objectsAreDeepEqual } = await import('../objectUtilities');
+
+      expect(objectsAreDeepEqual({ a: { b: 1 } }, { a: { b: 1 } })).toEqual(true);
+      expect(objectsAreDeepEqual({ a: { b: 1 } }, { a: { b: 2 } })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: { b: 1 } }, { a: { c: 1 } })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: { b: 1 } }, { a: { b: 1, c: 2 } })).toEqual(false);
+      expect(objectsAreDeepEqual({ a: { b: 1 } }, { a: { b: 1 }, c: 2 })).toEqual(false);
+    });
+  });
+
+  describe('cloneDeep', () => {
+    it('can clone primitives', async () => {
+      const { cloneDeep } = await import('../objectUtilities');
+
+      expect(cloneDeep(1)).toEqual(1);
+      expect(cloneDeep('a')).toEqual('a');
+      expect(cloneDeep(true)).toEqual(true);
+      expect(cloneDeep(undefined)).toEqual(undefined);
+      expect(cloneDeep(null)).toEqual(null);
+    });
+
+    it('can clone arrays', async () => {
+      const { cloneDeep } = await import('../objectUtilities');
+
+      expect(cloneDeep([])).toEqual([]);
+      expect(cloneDeep([1])).toEqual([1]);
+      expect(cloneDeep([1, 2])).toEqual([1, 2]);
+      expect(cloneDeep([1, 2, 3])).toEqual([1, 2, 3]);
+    });
+
+    it('can clone objects', async () => {
+      const { cloneDeep } = await import('../objectUtilities');
+
+      expect(cloneDeep({})).toEqual({});
+      expect(cloneDeep({ a: 1 })).toEqual({ a: 1 });
+      expect(cloneDeep({ a: 1, b: 1 })).toEqual({ a: 1, b: 1 });
+      expect(cloneDeep({ a: 1, b: 2 })).toEqual({ a: 1, b: 2 });
+
+      const a: Record<string, number> = { a: 1 };
+      expect(cloneDeep({ a, b: a })).toEqual({ a: { a: 1 }, b: { a: 1 } });
+    });
+
+    it('can clone nested objects', async () => {
+      const { cloneDeep } = await import('../objectUtilities');
+
+      expect(cloneDeep({ a: { b: 1 } })).toEqual({ a: { b: 1 } });
+    });
```

Comment:
All three of these tests should have checks that the resulting object is `!==` to the original.